polars/polars-arrow/src/kernels/rolling/no_nulls.rs

```diff
@@ -73,6 +88,79 @@ where
     ))
 }
 
+fn rolling_apply_quantile<T, Fo, Fa>(
+    values: &[T],
+    quantile: f64,
+    interpolation: QuantileInterpolOptions,
+    window_size: usize,
+    min_periods: usize,
+    det_offsets_fn: Fo,
+    aggregator: Fa,
+) -> ArrayRef
+where
+    Fo: Fn(Idx, WindowSize, Len) -> (Start, End),
+    Fa: Fn(&[T], f64, QuantileInterpolOptions) -> f64,
+    T: Debug,
+{
+    let len = values.len();
+    let out = (0..len)
+        .map(|idx| {
+            let (start, end) = det_offsets_fn(idx, window_size, len);
+            let vals = unsafe { values.get_unchecked(start..end) };
+            aggregator(vals, quantile, interpolation)
+        })
+        .collect_trusted::<Vec<f64>>();
+
+    let validity = create_validity(min_periods, len as usize, window_size, det_offsets_fn);
+    Arc::new(PrimitiveArray::from_data(
+        DataType::Float64,
+        out.into(),
+        validity.map(|b| b.into()),
+    ))
+}
+
+#[allow(clippy::too_many_arguments)]
+fn rolling_apply_weighted_quantile<T, K, Fo, Fa>(
+    values: &[T],
+    argument: f64,
+    interpolation: QuantileInterpolOptions,
+    window_size: usize,
+    min_periods: usize,
+    det_offsets_fn: Fo,
+    aggregator: Fa,
+    weights: &[f64],
+) -> ArrayRef
+where
+    Fo: Fn(Idx, WindowSize, Len) -> (Start, End),
+    Fa: Fn(&[(T, f64)], f64, QuantileInterpolOptions) -> K,
+    K: NativeType,
+    T: Debug + Copy,
+{
+    assert_eq!(weights.len(), window_size);
+    let mut buf: Vec<(T, f64)> = Vec::with_capacity(window_size);
+    let len = values.len();
+    let out = (0..len)
+        .map(|idx| {
+            let (start, end) = det_offsets_fn(idx, window_size, len);
+            let vals = unsafe { values.get_unchecked(start..end) };
+            buf = vals
+                .iter()
+                .zip(weights.iter())
+                .map(|(&x, &y)| (x, y))
```

Comment:
Is there a reason we don't multiply the weights here directly? Then we can take the same `Fa` as in `rolling_apply_weighted`:  `Fn(&[T], f64, QuantileInterpolOptions) -> f64`

Reply:
I interpreted weighted quantiles to be like [this definition](https://en.wikipedia.org/wiki/Weighted_median) of the weighted median. In this definition the order is preserved but the quantiles are just shifted around by the weights. Multiplying in the weights would potentially change the order. For example for values (1, 2, 3) and weights (0.97, 0.02, 0.01), if we multiply the weights in then the order reverses.

Happy to do it that way if you want but the use case for that is not clear to me

Reply:
The weights are intended to scale the window. For instance you could create a triangular window like so `1, 0.5, 0`. 

I think we should keep the weights interpretation the same for all rolling methods for consistency. 

We can add a weighted median/quantile later that takes separate weights for that operation.

Reply:
Okay great, will do

Reply:
this is done now